tests/test_Vehicle.cpp

```diff
@@ -95,18 +95,18 @@ BOOST_AUTO_TEST_CASE(test_hinges) {
 
     BOOST_REQUIRE(part != nullptr);
 
-    BOOST_CHECK_EQUAL(part->constraint, nullptr);
-    BOOST_CHECK_EQUAL(part->body, nullptr);
+    BOOST_CHECK_EQUAL(part->constraint.get(), nullptr);
+    BOOST_CHECK_EQUAL(part->body.get(), nullptr);
 
     vehicle->setPartLocked(part, false);
 
-    BOOST_CHECK_NE(part->body, nullptr);
-    BOOST_CHECK_NE(part->constraint, nullptr);
+    BOOST_CHECK_NE(part->body.get(), nullptr);
+    BOOST_CHECK_NE(part->constraint.get(), nullptr);
 
     vehicle->setPartLocked(part, true);
 
-    BOOST_CHECK_EQUAL(part->constraint, nullptr);
-    BOOST_CHECK_EQUAL(part->body, nullptr);
```

Comment:
Can this just be `BOOST_CHECK(part->body == nullptr)`?

Reply:
I've rearranged vehicle tests a bit now.